step-plans/step-plans-core/src/main/java/step/core/artefacts/handlers/ArtefactHandler.java

```diff
@@ -114,11 +114,12 @@ public void createReportSkeleton(ReportNode parentReportNode, ARTEFACT artefact,
 		}
 
 		try {
-			dynamicBeanResolver.evaluate(artefact, getBindings());
+			dynamicBeanResolver.evaluateMandatoryFieldsEvenForSkippedArtefact(artefact, getBindings());

```

Comment:
If the artefact is not skipped, the method getBindings() and the evaluation of DynamicName and SkipNode are now evaluated twice. To avoid this, I would move this line to the line 120